### ISSUE TITLE
Firebase Messaging deleteInstanceID() Permanently disables notifications

### ISSUE THREAD

**Issue by adminxsoteria** [flutter/flutter#29964]
Hi Flutter Developers!

I'm running into trouble implementing push notifications on iOS using FCM, specifically enabling and disabling them. In order to disable notifications for a given device, I run this code...
```dart
await firebaseMessaging.setAutoInitEnabled(false);
firebaseMessaging.deleteInstanceID();
```
The above code revokes any tokens that have been generated for this device. It works very well!

In order to reenable push notifications, I run this code...
```dart
firebaseMessaging.setAutoInitEnabled(true);
```
This code doesn't work so well. Allow me to explain...

The first time I build and load the app, I can send a notification using the firebase console and everything works perfectly. After I call `firebaseMessaging.deleteInstanceID()` I don't get notifications sent from the console (which is as it should be). My question is how can I basically undo that call so that a device can _once again_ receive notifications. I thought that a simple `firebaseMessaging.setAutoInitEnabled(true)` call would do it, but it looks like that's not the I case. I think this is a very important use case for any sort of app with user authentication (i.e. enable and disable notifications based on if the user is signed in or not). 

The docs for `.disableInstanceID()` read `Resets Instance ID and revokes all tokens. In iOS, it also unregisters from remote notifications.`. That `unregisters from remote notifications` part I believe is what I would like to reverse in order to reenable notifications.

Something weird I found is that if I close the app (as in kill it by double tapping home and swiping up) and then reload the app, I can once again receive notifications.

Please be advised that I experience the same behavior when sending notifications to a specific device by using a token.

**Comment by adriancmurray** [comment]
You'll likely want to check out this response on SO: https://stackoverflow.com/a/54491607/9910719

TL;DR, don't call deleteInstanceID().

**Comment by hebersousa** [comment]
I'm facing the same problem.
I do not need to keep the same registration token. so I'm using deleteInstanceID () on logout. But after calling deleteInstanceID () I no longer get any token with getToken (). I realized that this only happens if I call configure () before. This problem does not occur in the ios emulator and Android device.
How to solve?

Iphone 6s, IOS 12.3.1
Flutter doctor
`[✓] Flutter (Channel stable, v1.5.4-hotfix.2, on Mac OS X 10.14.5 18F132, locale
    pt-BR)
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] iOS toolchain - develop for iOS devices (Xcode 10.2.1)
[✓] Android Studio (version 3.4)`

**Comment by escamoteur** [comment]
Hi, if you have further questions on this problem, please open a new issue at Firebase's new home at https://github.com/FirebaseExtended/flutterfire
